File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int curr = 0;
        int min = prices[0];
        
        for (int i=0; i<prices.size(); ++i) {
            if (prices[i] < min)
                min = prices[i];
            
            curr = prices[i] - min;
            if (curr > maxProfit) {
                maxProfit = curr;
            }
        }
        
        return maxProfit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minSoFar = prices[0]
        greatestDiff = 0
        
        for p in prices:
            if p < minSoFar:
                minSoFar = p
            
            if p - minSoFar > greatestDiff:
                greatestDiff = p - minSoFar
        
        return greatestDiff